CI.hs

```diff
@@ -86,15 +93,21 @@ main = do
         putStrLn $ "# Completed in " ++ showDuration t ++ ": " ++ x ++ "\n"
         hFlush stdout
 
-      mkTarball :: IO String
-      mkTarball = do
+      mkTarball :: String -> IO ()
+      mkTarball target = do
         stackYaml <- readFile' "stack.yaml"
         writeFile "stack.yaml" $ stackYaml ++ unlines ["- ghc"]
-        sDistCreateExtract
+        sDistCreateExtract target
 
-      sDistCreateExtract :: IO String
-      sDistCreateExtract = do
+      sDistCreateExtract :: String -> IO ()
+      sDistCreateExtract target = do
         cmd "stack sdist ghc --tar-dir=."
-        [tarball] <- filter (isExtensionOf ".tar.gz") <$> getDirectoryContents "."
-        cmd $ "tar -xf " ++ tarball
-        return tarball
+        cmd $ "tar -xvf " ++ target ++ ".tar.gz"
+
+      rmFiles :: [FilePath] -> IO ()
+      rmFiles fs = forM_ fs $
+        \f -> removeFile f `catchIOError` (const  $ return ())
```

Comment:
Can you check if the file exists and then remove it instead of catching the IO exception?

Reply:
It's not worth it. The intent is just try and remove it and ignore any failure.